### PR TITLE
oop-special_class_method-hard-q9

### PR DESCRIPTION
**Pull request by KaiKhademi**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [x]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [x]  Do the comments for Parsons follow the correct convention?
- [x]  Is the question being tested sufficiently?
    - [x]  Are there representative cases to test the program?
    - [x]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”

### CODE REVIEW

File: src/oop/special_class_method/hard/q9/QuestionText.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# oop-special_class_method-hard-q9
+
+## Question Text
+
+Many people are fascinated by our planet's solar system. Make a Java class called **Earth** with two attributes; a
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/QuestionText.md:5]: A lot of the text here is made redundant by the class description below

File: src/oop/special_class_method/hard/q9/QuestionText.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    - An `int` attribute named **population** which is not accessible from other classes
+
+- Constructors:
+    - A constructor which is accessible from other classes and accepts a`double` and a `String` to initialize the
+      **landMass** and **population** attributes.
+
+- Methods:
+    - a `setLandMass` method which accepts a `double` value, updates the **landMass** attribute with it, and is
+      accessible
+      from other classes
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/QuestionText.md:26]: fix formatting

File: src/oop/special_class_method/hard/q9/QuestionText.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    - a `getLandMass` method which is accessible from other classes and returns the value of the **landMass** attribute
+    - a `setPopulation` method which accepts an `int` value, updates the **population** attribute with it, and is
+      accessible
+      from other classes
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/QuestionText.md:30]: fix formatting

File: src/oop/special_class_method/hard/q9/QuestionText.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    - a `getPopulation` method which is accessible from other classes and returns the value of the **population**
+      attribute
+    - a `toString` method which returns a String of the information about the `Earth` class of the form
+      `"Here " + population + "people live on a " + landMass + "kg planet."`.
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/QuestionText.md:34]: change the description formatting to be more like what was discussed in meeting but the rest of the toString stuff in the question can stay the same

File: src/oop/special_class_method/hard/q9/QuestionText.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+### Distractors:
+
+- setLandMass( lm );
+- setPopulation( p );
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/QuestionText.md:39]: these first two distractors don't really work I think since they could theoretically be used in the constructor.

File: src/oop/special_class_method/hard/q9/QuestionText.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+- return Population;
+- return population = a;
+- public int population;
+- public double landMass;
+- private class Earth
+- public Earth()
+- int population;
+- double landMass;
+- public String toString()
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/QuestionText.md:48]: depending how the question is input in the website this could not work as a distractor. When I input questions I like to separate brackets from the line they're attached to in which case this would just become a duplicate line.

File: src/oop/special_class_method/hard/q9/QuestionText.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+- public static void setLandMass(double a)
+- private Earth( double landMass, int population)
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/QuestionText.md:50]: add a distractor which is a constructor that is (int double)

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+    public void earthClassHasCorrectConstructor(double value1, int value2) throws Throwable {
+        Object[][] arguments = {
+                {value1, double.class},
+                {value2, int.class}
+        };
+        Class<?>[] constructorClasses = {double.class, int.class};
+        Object classInstance = testClass.createInstance(arguments);
+        String incorrectDoubleValueMessage =
+                "Your " + className + " constructor does not initialize the " + doubleFieldName + " attribute to the correct value.";
+        String incorrectStringValueMessage =
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/MainTest.java:69]: this should not be labelled string

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+                {initialValue, double.class},
+                {0, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        Object[][] setSizeArguments = {
+                {setValue, double.class}
+        };
+        String[] setMethodModifiers = {"public"};
+        String incorrectSetMethodMessage = String.join(" ",
+                "Your", setDoubleMethodName, "does not correctly set the value of the", doubleFieldName, "attribute.");
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/MainTest.java:103]: in this message and any other make sure that after a method name there is "method"

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 10]
```diff
@@ -0,0 +1,216 @@
...
+
+    @ParameterizedTest
+    @MethodSource("twoIntInputProvider")
+    public void earthClassHasCorrectSetPopulationMethod(int initialValue, int updatedValue) throws Throwable {
+        Object[][] arguments = {
+                {Math.random() * 1000, double.class},
+                {initialValue, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        testClass.setFieldValue(classInstance, initialValue, intFieldName);
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/MainTest.java:141]: why is this here?

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 11]
```diff
@@ -0,0 +1,216 @@
...
+
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    public void earthClassHasCorrectGetPopulationMethod(double value1, int value2) throws Throwable {
+        Object[][] arguments = {
+                {value1, double.class},
+                {value2, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        testClass.setFieldValue(classInstance, value2, intFieldName);
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/MainTest.java:161]: why is this here?

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 12]
```diff
@@ -0,0 +1,216 @@
...
+                "Your " + getDoubleMethodName + " method does not return the correct value.");
+        _assertEquals(0, testClass.callMethod(getIntMethodName, classInstance),
+                "Your " + getIntMethodName + " method does not return the correct value.");
+        Object[][] setDoubleArguments = {
+                {value, double.class}
+        };
+        testClass.callMethod(setDoubleMethodName, setDoubleArguments, classInstance);
+        _assertEquals(value, testClass.callMethod(getDoubleMethodName, classInstance),
+                "Your " + getDoubleMethodName + " method does not return the correct value calling the " + setDoubleMethodName + " method.");
+        Object[][] setBooleanArguments = {
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/MainTest.java:204]: this should not be labelled boolean

**Review by SamamthaH — CHANGES_REQUESTED** [review]

File: src/oop/special_class_method/hard/q9/QuestionText.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+- Methods:
+    - a `setLandMass` method which accepts a `double` value, updates the **landMass** attribute with it, and is
+      accessible from other classes
+    - a `getLandMass` method which is accessible from other classes and returns the value of the **landMass** attribute
+    - a `setPopulation` method which accepts an `int` value, updates the **population** attribute with it, and is
+      accessible from other classes
+    - a `getPopulation` method which is accessible from other classes and returns the value of the **population**
+      attribute
+    - a `toString` method which returns a String of the information about the `Earth` class of the form
+      `"Here " + x + "people live on a " + y + "kg planet."` where x is the value of **population** attribute and
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/QuestionText.md:29]:
```suggestion
      `"Here x people live on a ykg planet."` where x is the value of **population** attribute and
```

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 14]
```diff
@@ -0,0 +1,210 @@
...
+        Object classInstance = testClass.createInstance(arguments);
+        Object[][] setMethodArguments = {
+                {updatedValue, int.class}
+        };
+        String[] setMethodModifiers = {"public"};
+        String incorrectSetMethodMessage = String.join(" ",
+                "Your", setIntMethodName, "method does not correctly set the value of the", intFieldName, "attribute.");
+        Object setMethodOutput = testClass.callMethod(setIntMethodName, setMethodArguments, setMethodModifiers, classInstance);
+        _assertEquals(updatedValue, testClass.getFieldValue(classInstance, intFieldName), incorrectSetMethodMessage);
+        assertNull(setMethodOutput, String.join(" ", "Your", setIntMethodName, "should not return any output"));
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/MainTest.java:145]:
```suggestion
        assertNull(setMethodOutput, String.join(" ", "Your", setIntMethodName, "method should not return any output"));
```

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 15]
```diff
@@ -0,0 +1,210 @@
...
+        Object classInstance = testClass.createInstance(arguments);
+        Object[][] setSizeArguments = {
+                {setValue, double.class}
+        };
+        String[] setMethodModifiers = {"public"};
+        String incorrectSetMethodMessage = String.join(" ",
+                "Your", setDoubleMethodName, "method does not correctly set the value of the", doubleFieldName, "attribute.");
+        Object setMethodOutput = testClass.callMethod(setDoubleMethodName, setSizeArguments, setMethodModifiers, classInstance);
+        _assertEquals(setValue, testClass.getFieldValue(classInstance, doubleFieldName), incorrectSetMethodMessage);
+        assertNull(setMethodOutput, String.join(" ", "Your", setDoubleMethodName, "should not return any output"));
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/MainTest.java:102]:
```suggestion
        assertNull(setMethodOutput, String.join(" ", "Your", setDoubleMethodName, "method should not return any output"));
```

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 16]
```diff
@@ -0,0 +1,210 @@
...
+    public void earthClassHasCorrectToStringMethod(double value, int b) throws Throwable {
+        Object[][] arguments = {
+                {value, double.class},
+                {b, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        String[] methodModifiers = {"public"};
+        String expected = "Here " + b + "people live on a " + value + "kg planet.";
+        String incorrectToStringMessage = String.join(" ",
+                "Your", className, " toString method does not return the correct String.");
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/MainTest.java:174]:
```suggestion
                "Your", className, "toString method does not return the correct String.");
```

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 17]
```diff
@@ -0,0 +1,210 @@
...
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    public void earthClassMethodsWorkTogether(double value, int b) throws Throwable {
+        double initialValue = Math.random() * 100;
+        Object[][] arguments = {
+                {initialValue, double.class},
+                {0, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        _assertEquals(initialValue, testClass.callMethod(getDoubleMethodName, classInstance),
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/MainTest.java:188]: no need to test get methods before set methods are called

**Review by SamamthaH — CHANGES_REQUESTED** [review]

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 18]
```diff
@@ -0,0 +1,210 @@
...
+    private final String setDoubleMethodName = "setLandMass";
+    private ObjectTest testClass;
+
+    @BeforeEach
+    public void setUp() {
+        String classString = "oop.special_class_method.hard.q9." + className;
+        testClass = new ObjectTest(classString);
+    }
+
+    @Test
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/MainTest.java:32]: Modify to include the new hasField method

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 19]
```diff
@@ -0,0 +1,210 @@
...
+    public void earthClassHasCorrectAttributes() {
+        assertTrue(testClass.hasField(doubleFieldName, double.class),
+                "Your " + className + " class is missing the " + doubleFieldName + " attributes.");
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/MainTest.java:35]:
```suggestion
                "Your " + className + " class is missing the " + doubleFieldName + " attribute.");
```

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 20]
```diff
@@ -0,0 +1,210 @@
...
+        assertTrue(testClass.hasModifier(doubleFieldName, "private"),
+                "Your " + doubleFieldName + " does not have the correct visibility modifier.");
+        assertTrue(testClass.hasField(intFieldName, int.class),
+                "Your " + className + " class is missing the " + intFieldName + " attributes.");
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/MainTest.java:39]:
```suggestion
                "Your " + className + " class is missing the " + intFieldName + " attribute.");
```

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 21]
```diff
@@ -0,0 +1,210 @@
...
+    private static Stream<Arguments> constructorInputProvider() {
+        return Stream.of(
+                Arguments.of(4535.4, 300),
+                Arguments.of(234.4, 100),
+                Arguments.of(654.45, 365),
+                Arguments.of(43.3, 14)
+
+        );
+    }
+
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/MainTest.java:53]: There is no assert to check if the constructor exists, only that is initializes attributes to the correct value and then that it has the correct visibility modifier

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 22]
```diff
@@ -0,0 +1,210 @@
...
+        _assertEquals(initialValue, testClass.callMethod(getDoubleMethodName, classInstance),
+                "Your " + getDoubleMethodName + " method does not return the correct value.");
+        _assertEquals(0, testClass.callMethod(getIntMethodName, classInstance),
+                "Your " + getIntMethodName + " method does not return the correct value.");
+        Object[][] setDoubleArguments = {
+                {value, double.class}
+        };
+        testClass.callMethod(setDoubleMethodName, setDoubleArguments, classInstance);
+        _assertEquals(value, testClass.callMethod(getDoubleMethodName, classInstance),
+                "Your " + getDoubleMethodName + " method does not return the correct value calling the " + setDoubleMethodName + " method.");
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/MainTest.java:197]:
```suggestion
                "Your " + getDoubleMethodName + " method does not return the correct value after calling the " + setDoubleMethodName + " method.");
```

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 23]
```diff
@@ -0,0 +1,210 @@
...
+        Object[][] setArguments = {
+                {b, int.class}
+        };
+        testClass.callMethod(setIntMethodName, setArguments, classInstance);
+        _assertEquals(b, testClass.callMethod(getIntMethodName, classInstance),
+                "Your " + getIntMethodName + " method does not return the correct value calling the " + setIntMethodName + " method.");
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/MainTest.java:203]:
```suggestion
                "Your " + getIntMethodName + " method does not return the correct value after calling the " + setIntMethodName + " method.");
```

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 24]
```diff
@@ -0,0 +1,210 @@
...
+        String expectedToString = "Here " + b + "people live on a " + value + "kg planet.";
+        String incorrectToStringMessage = String.join(" ",
+                "Your", className, " toString method does not return the correct String after updating the values of its attributes using its setter methods.");
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/MainTest.java:206]:
```suggestion
                "Your", className, "toString method does not return the correct String after updating the values of its attributes using its setter methods.");
```

File: src/oop/special_class_method/hard/q9/QuestionText.md [hunk 25]
```diff
@@ -0,0 +1,44 @@
...
+### Earth Class:
+
+Your `Earth` class must have the following items:
+
+- Attributes:
+    - A `double` attribute named **landMass** which is not accessible from other classes
+    - An `int` attribute named **population** which is not accessible from other classes
+
+- Constructors:
+    - A constructor which is accessible from other classes and accepts a`double` and a `int` to initialize the
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/QuestionText.md:17]:
```suggestion
    - A constructor which is accessible from other classes and accepts a `double` and an `int` to initialize the
```

File: src/oop/special_class_method/hard/q9/QuestionText.md [hunk 26]
```diff
@@ -0,0 +1,44 @@
...
+- return Population;
+- return population = a;
+- public int population;
+- public double landMass;
+- private class Earth
+- public Earth()
+- int population;
+- double landMass;
+- public static void setLandMass(double a)
+- private Earth( double landMass, int population)
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/QuestionText.md:43]:
```suggestion
- private Earth(double landMass, int population)
```

**Review by SamamthaH — CHANGES_REQUESTED** [review]

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 27]
```diff
@@ -0,0 +1,219 @@
...
+                {0, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        Object[][] setSizeArguments = {
+                {setValue, double.class}
+        };
+        String[] setMethodModifiers = {"public"};
+        String incorrectSetMethodMessage = String.join(" ",
+                "Your", setDoubleMethodName, "method does not correctly set the value of the", doubleFieldName, "attribute.");
+        Object setMethodOutput = testClass.callMethod(setDoubleMethodName, setSizeArguments, setMethodModifiers, classInstance);
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/MainTest.java:109]: use .hasMethod first with void.type instead of assertNull here and everywhere else

File: src/oop/special_class_method/hard/q9/QuestionText.md [hunk 28]
```diff
@@ -0,0 +1,44 @@
+# oop-special_class_method-hard-q9
+
+## Question Text
+
+Many people are fascinated by our planet's solar system. Make a Java class called **Earth** with two attributes; a
+`double` **landMass** and an `int` named **population**.
```
Comment (SamamthaH) [inline src/oop/special_class_method/hard/q9/QuestionText.md:6]: get rid of duplicate class information

**Review by SamamthaH — CHANGES_REQUESTED** [review]

File: src/oop/special_class_method/hard/q9/QuestionText.md [hunk 29]
```diff
@@ -0,0 +1,39 @@
...
+    - a `getLandMass` method which is accessible from other classes and returns the value of the **landMass** attribute
+    - a `setPopulation` method which accepts an `int` value, updates the **population** attribute with it, and is
+      accessible from other classes
+    - a `getPopulation` method which is accessible from other classes and returns the value of the **population**
+      attribute
+    - a `toString` method which returns a String of the information about the `Earth` class of the form
+      `"Here x people live on a ykg planet."` where x is the value of **population** attribute and
+      y is the value of **landMass** attribute.
+
+### Distractors:
```
[the remark this answers is not in the document]
Comment (KaiKhademi) [reply src/oop/special_class_method/hard/q9/QuestionText.md:27]: Yes that is also mentioned in the MainTest class

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 30]
```diff
@@ -0,0 +1,289 @@
...
+        };
+        String[] setMethodModifiers = {"public"};
+        String incorrectSetMethodMessage = String.join(" ",
+                "Your", setDoubleMethodName, "method does not correctly set the value of the", doubleFieldName, "attribute.");
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(setDoubleMethodName, new Class<?>[]{double.class}), String.format(incorrectMethodDefinition, setDoubleMethodName, className));
+        assertTrue(testClass.hasModifier(setDoubleMethodName, new Class<?>[]{double.class}, "public"), String.format(incorrectModifierMessage, setDoubleMethodName, className));
+        assertTrue(testClass.hasReturnType(setDoubleMethodName, new Class<?>[]{double.class}, Void.TYPE), String.format(incorrectReturnType, setDoubleMethodName, className));
```
Comment (Hedgemon4) [inline src/oop/special_class_method/hard/q9/MainTest.java:124]: These should come first. Also, if this is a parsons question I don't think we should have two separate tests, it can just be one.

Comment (Hedgemon4) [reply src/oop/special_class_method/hard/q9/MainTest.java:124]: You can apply this for the other methods below

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/oop/special_class_method/hard/q9/QuestionText.md [hunk 31]
```diff
@@ -0,0 +1,45 @@
+# oop-special_class_method-hard-q9
+
+### Question Text
+
+Many people are fascinated by our planet's solar system. Make a Java class called Earth that represents one of those
+planets with the requirements below:
```
Comment (Hedgemon4) [inline src/oop/special_class_method/hard/q9/QuestionText.md:6]:
```suggestion
Many people are fascinated by our planet's Solar System. Make a Java class called Earth that models one of the planets in our Solar System with the requirements below:
```

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 32]
```diff
@@ -0,0 +1,287 @@
...
+                "Your " + intFieldName + " does not have the correct visibility modifier.");
+    }
+
+    private static Stream<Arguments> constructorInputProvider() {
+        return Stream.of(
+                Arguments.of(4535.4, 300),
+                Arguments.of(234.4, 100),
+                Arguments.of(654.45, 365),
+                Arguments.of(43.3, 14)
+
```
Comment (Hedgemon4) [inline src/oop/special_class_method/hard/q9/MainTest.java:54]: Should have constructor with zero as the input

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 33]
```diff
@@ -0,0 +1,287 @@
...
+                incorrectIntValueMessage);
+    }
+
+    private static Stream<Arguments> doubleInputProvider() {
+        return Stream.of(
+                Arguments.of(4535.4, 432.2),
+                Arguments.of(234.4, 655.3),
+                Arguments.of(654.45, 453.3),
+                Arguments.of(43.3, 654.2)
+
```
Comment (Hedgemon4) [inline src/oop/special_class_method/hard/q9/MainTest.java:91]: Add input provider where value is changed to zero.

File: src/oop/special_class_method/hard/q9/MainTest.java [hunk 34]
```diff
@@ -0,0 +1,287 @@
...
+        Object getMethodOutput = testClass.callMethod(getDoubleMethodName, getMethodModifiers, classInstance);
+        _assertEquals(value1, getMethodOutput, incorrectGetMethodMessage);
+    }
+
+    private static Stream<Arguments> twoIntInputProvider() {
+        return Stream.of(
+                Arguments.of(10, 15),
+                Arguments.of(30, 10),
+                Arguments.of(100, 120),
+                Arguments.of(1, 2)
```
Comment (Hedgemon4) [inline src/oop/special_class_method/hard/q9/MainTest.java:165]: Same thing here, we should add a test for setting a non-zero value to zero.

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]
Just a few small things to change and then this should be good to merge.

**Review by Hedgemon4 — APPROVED** [review]
